colorblindfriendly.py

```diff
@@ -63,42 +74,138 @@
-cb_colors = (
-    ("black", (0.000, 0.000, 0.000),                   # (  0,   0,   0)
-            ()),
-    ("orange", (0.902, 0.624, 0.000),                   # (230, 159,   0)
-     ()),
-    ("sky_blue", (0.337, 0.706, 0.914),                   # ( 86, 180, 233)
-     ("skyblue", "light_blue", "lightblue")),
-    ("bluish_green", (0.000, 0.620, 0.451),                   # (  0, 158, 115)
-     ("bluishgreen", "green")),
-    ("yellow", (0.941, 0.894, 0.259),                   # (240, 228,  66)
-     ()),
-    ("blue", (0.000, 0.447, 0.698),                   # (  0, 114, 178)
-     ()),
-    ("vermillion", (0.835, 0.369, 0.000),                   # (213,  94,   0)
-     ("red", "red_orange", "redorange")),
-    ("reddish_purple", (0.800, 0.475, 0.655),                   # (204, 121, 167)
-     ("reddishpurple", "rose", "violet", "magenta")),
-)
-
-for c in cb_colors:
-    # main name
-    cmd.set_color("cb_%s" % c[0], c[1])
-    print("Set color: cb_%s" % c[0])
-
-    # alternate names
-    for alt in c[2]:
-        cmd.set_color("cb_%s" % alt, c[1])
-        print("           cb_%s" % alt)
+CB_COLORS = {
+    'black': {
+        'rgb': [0, 0, 0],
+        'alt': None,
+    },
+    'orange': {
+        'rgb': [230, 159, 0],
+        'alt': None,
+    },
+    'sky_blue': {
+        'rgb': [86, 180, 233],
+        'alt': ['skyblue', 'light_blue', 'lightblue'],
+    },
+    'bluish_green': {
+        'rgb': [0, 158, 115],
+        'alt': ['bluishgreen', 'green'],
+    },
+    'yellow': {
+        'rgb': [240, 228, 66],
+        'alt': None,
+    },
+    'blue': {
+        'rgb': [0, 114, 178],
+        'alt': None,
+    },
+    'vermillion': {
+        'rgb': [213, 94, 0],
+        'alt': ['red', 'red_orange', 'redorange'],
+    },
+    'reddish_purple': {
+        'rgb': [204, 121, 167],
+        'alt': ['reddishpurple', 'rose', 'violet', 'magenta'],
+    },
+}
+
+
+def set_colors(replace=False):
+    '''Add the color blind-friendly colors to PyMOL.'''
+    # Track the added colors
+    added_colors = []
+
+    for color, properties in CB_COLORS.items():
+        # RGB tuple shortcut
+        rgb = properties['rgb']
+
+        # Get the primary and alternate color names into a single list
+        names = [color]
+        if properties['alt']:
+            names.extend(properties['alt'])
+
+        # Set the colors
+        for name in names:
+            # Set the cb_color
+            cb_name = 'cb_{}'.format(name)
+            cmd.set_color(cb_name, rgb)
+
+            # Optionally replace built-in colors
+            if replace:
+                cmd.set_color(name, rgb)
+                spacer = (20 - len(name)) * ' '
+                added_colors.append('{}{}{}'.format(name, spacer, cb_name))
+            else:
+                added_colors.append(cb_name)
+
+    # Notify user of newly available colors
+    print('\nColor blind-friendly colors are now available:')
+    print(textwrap.indent('\n'.join(added_colors), '    '))
```

Comment:
`textwrap.indent` not available in Python 2. I suggest to simply add the spaces in lines 159 and 161 when appending to `added_colors`.

Reply:
Thanks, done.